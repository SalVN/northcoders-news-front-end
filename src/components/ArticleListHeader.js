import React from 'react';
import PropTypes from 'prop-types';

import './css/MainArticleList.css';

const ArticleListHeader = function (props) {
    return (
        <div className='content'>
            <div className='columns'>
                <div className='column is-3'>
                    <h2 className='page-title'><strong>{props.title}</strong></h2>
                </div>
                <div className='column is-1 is-offset-5 label'>
                    <label htmlFor="Sort by">Sort by:</label>
                </div>
                <div className='column is-3'>
                    <div className='select is-primary select-menu'>
                        <select name="Sort by" onChange={props.handleClickSelect}>
                            <option value="votes">Popularity</option>
                            <option value="comments">Number of Comments</option>
                            <option value="random">Random</option>
                        </select>
                    </div>
                </div>
            </div>
        </div>
    );
};

ArticleListHeader.propTypes = {
    title: PropTypes.string.isRequired,
    handleClickSelect: PropTypes.func.isRequired
};

export default ArticleListHeader;