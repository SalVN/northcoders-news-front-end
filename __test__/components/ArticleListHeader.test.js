import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';


import ArticleListHeader from '../../src/components/ArticleListHeader';

describe('ArticleListHeader', () => {
    it('is a function', () => {
        expect(typeof ArticleListHeader).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<ArticleListHeader
            title={'abcdefg'}
            handleClickSelect={x => x}
            toggleDropdown={x => x}
            showDropdown={false}
        />);
        expect(enzymeWrapper.children().length).toEqual(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ArticleListHeader
                    title={'abcdefg'}
                    handleClickSelect={x => x}
                    toggleDropdown={x => x}
                    showDropdown={false} />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});