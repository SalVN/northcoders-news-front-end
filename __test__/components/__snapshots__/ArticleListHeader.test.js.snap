// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListHeader renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <h2
        className="page-title"
      >
        <strong>
          abcdefg
        </strong>
      </h2>
    </div>
    <div
      className="column label has-text-left-mobile has-text-right-tablet"
    >
      <div
        className="dropdown has-text-left "
      >
        <div
          className="dropdown-trigger"
        >
          <button
            aria-controls="dropdown-menu"
            aria-haspopup="true"
            className="button dropdown-button-article-list"
            onClick={[Function]}
          >
            <label
              htmlFor="Sort by"
            >
              Sort by:
            </label>
            <span
              className="icon"
            >
              <i
                aria-hidden="true"
                className="fa fa-caret-down"
              />
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu"
          id="dropdown-menu"
          role="menu"
        >
          <div
            className="dropdown-content"
          >
            <a
              className="dropdown-item popular"
              onClick={[Function]}
              value="votes"
            >
              Popularity
            </a>
            <a
              className="dropdown-item comments"
              onClick={[Function]}
              value="comments"
            >
              Number of Comments
            </a>
            <a
              className="dropdown-item random"
              onClick={[Function]}
              value="random"
            >
              Random
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
